test/DelegationRegistryWithMetaTx.t.sol

```diff
@@ -0,0 +1,539 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import {Test} from "forge-std/Test.sol";
+import {console2} from "forge-std/console2.sol";
+import {SigUtils} from "test/utils/SigUtils.sol";
+
+import {DelegationRegistry} from "src/DelegationRegistry.sol";
+import {DelegationRegistryForwarder} from "src/DelegationRegistryForwarder.sol";
+import {IDelegationRegistry} from "src/IDelegationRegistry.sol";
+import {MinimalForwarder} from "openzeppelin-contracts/contracts/metatx/MinimalForwarder.sol";
+
+contract DelegationRegistrWithMetaTxTest is Test {
+    DelegationRegistryForwarder fwd;
+    DelegationRegistry reg;
+    SigUtils sigUtils;
+
+    address internal vault;
+    address internal delegate;
+    mapping(address => uint256) privateKeys;
+
+    function setUp() public {
+        fwd = new DelegationRegistryForwarder();
+        reg = new DelegationRegistry(address(fwd));
+        sigUtils = new SigUtils(fwd.domainSeparator());
+
+        uint256 vaultPrivateKey = 0xC11CE;
+        vault = vm.addr(vaultPrivateKey);
+        privateKeys[vault] = vaultPrivateKey;
```

Comment:
is there a better way to do this? we need private keys to do signing as a specific address, so we start with the key and use the derived address throughout the test suite